lua/lspconfig/server_configurations/drools_lsp.lua

```diff
@@ -1,35 +1,60 @@
 local util = require 'lspconfig.util'
 
-local get = function(config, section, key)
-  return vim.tbl_get(config.settings, section, key) and config.settings[section][key] or nil
+local function get(config, segments)
+  if vim.tbl_get(config, unpack(segments)) then
+    local cfg = config
+    for _, seg in ipairs(segments) do
+      cfg = cfg[seg]
+    end
+    return cfg
+  end
+  return nil
 end
 
-local get_java_bin = function()
-  local java_bin = vim.env.JAVA_HOME and util.path.join(vim.env.JAVA_HOME, 'bin', 'java') or 'java'
-  if vim.fn.has 'win32' == 1 then
-    java_bin = java_bin .. '.exe'
+local function get_java_bin(config)
+  local java_bin = get(config, { 'drools', 'java', 'bin' })
+  if java_bin == nil then
+    java_bin = vim.env.JAVA_HOME and util.path.join(vim.env.JAVA_HOME, 'bin', 'java') or 'java'
+    if vim.fn.has 'win32' == 1 then
+      java_bin = java_bin .. '.exe'
+    end
   end
   return java_bin
 end
 
+local function get_java_opts(config)
+  local java_opts = get(config, { 'drools', 'java', 'opts' })
+  return java_opts and java_opts or {}
+end
+
+local function get_jar(config)
+  local drools_jar = get(config, { 'drools', 'jar' })
+  if drools_jar == nil then
+    drools_jar = 'drools-lsp-server-jar-with-dependencies.jar'
+  end
+  return drools_jar
+end
+
+local function get_cmd(config)
+  local cmd = get(config, { 'cmd' })
+  if cmd == nil then
+    cmd = { get_java_bin(config) }
+    for _, o in ipairs(get_java_opts(config)) do
```

Comment:
```suggestion
    for _, o in ipairs(get(config, { 'drools', 'java', 'opts' })) do
```

Reply:
@glepnir My opinion is that it's more readable and maintainable to have a separate function for each config value we're looking for, and simply aggregate them into the resulting `cmd` in `get_cmd`. Is this a show-stopper?